### DISCUSSION TITLE
Partial label training on mask r-cnn

### DISCUSSION THREAD

**Discussion by HuangBugWei** [facebookresearch/detectron2#4970] [Q&A]
currently i have a bunch of partial labeled images, since RPN may predict the correct answer but the model may treat it as wrong answer due to partial labeled. How to deal with it?
i found out one solution might be setting ignore index in loss fn, but source code in detectron2 is using TORCH.NN.FUNCTIONAL.BINARY_CROSS_ENTROPY_WITH_LOGITS which is not support ignore index. plus it is not easy to modified the loss fn since it is integrated in model.
other possible solution is restricting RPN region, but still it is difficult to me to modified the code.
so i want to ask if there is a feature can solve this problem or any better suggestion even that i have misunderstanding to Mask R-CNN.

**Comment by HuangBugWei** [comment]
after delving deeper into the source code, i thought the function can be achieved by modifying the method label_and_sample_anchors from [https://github.com/facebookresearch/detectron2/blob/main/detectron2/modeling/proposal_generator/rpn.py#L307](url), add argument of images.labeledArea (modified dataloader) and check if anchor is out of interested part (the region out of labeled area since training data is partial labeled), i guess this is feasible and reasonable method right?
